src/dice.class.php

```diff
@@ -72,10 +75,12 @@ function requireMailsAreRegistered(array $emails) {
 	 */
 	function checkIfMailIsRegistered($email) {
 		$this->connectDatabase();
-
-		$sql = "SELECT registered_email FROM dice_emails WHERE registered_email = '$email'";
-		$result = $this->dbconn->query($sql) or exit("fatal error: data connection error {$this->dbconn->error}!");
-		return $result->num_rows === 1;
+		$statement = $this->dbconn->prepare("SELECT COUNT(*) FROM dice_emails WHERE registered_email=?");
+		$statement->bind_param("s", $email);
+		$statement->execute() or exit("fatal error: data connection error {$this->dbconn->error}!");
+		$statement->bind_result($email_count);
+		$statement->fetch();
+		return $email_count === 1;
```

Comment:
Looking at this again, it really seems this is just duplicating code.
Task for another PR.